subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc

```diff
@@ -49,27 +49,21 @@ This is a potentially breaking change if you are consuming the console output of
 
 === Deprecations
 
-[[custom_configuration_roles]]
-==== Custom configuration roles
-
-// TODO: There is currently no API for users to provide "custom roles" or even
-// any roles at all other than the existing role-mutating methods. We should revisit
-// this deprecation log and section when we introduce a public API for creating
-// locked-role configurations.
-
-Custom roles have been deprecated.
-Use a pre-defined role instead.
-
 [[configurations_allowed_usage]]
 ==== Allowed configurations usage
 
-The usage of configurations should be fixed at creation.
-Mutating the allowed usage on a configuration is deprecated.
+The allowed usage of a configuration should be immutable after creation.
```

Comment:
```suggestion
In general, the allowed usage of a configuration should be immutable after creation.
```
Added "In general" to signal that this is generally a good thing to do, but confirm that this isn't something set in stone.

Reply:
I disagree - our eventual goal is to make it set in stone.  I think "should be" vs. "is now" captures this.